cmd/object-api-interface.go

```diff
@@ -100,12 +105,23 @@ func (r *HashReader) Read(p []byte) (n int, err error) {
 	if err != nil && err != io.EOF {
```

Comment:
you would need to handle `io.UnexpectedEOF` something like
```go
if err == nil || err == io.UnexpectedEOF {
 // update md5 and sha256 hashers
}

if err == io.EOF || err == io.UnexpectedEOF {
//  call HashReader.Verify()
}

return
```

Reply:
Read() never returns io.UnexpectedEOF @balamurugana - UnexpectedEOF is only a higher level construct.